distribution/lib/Standard/Microsoft/0.0.0-dev/src/Internal/SQLServer_Dialect.enso

```diff
@@ -0,0 +1,720 @@
+import Standard.Base.Errors.Unimplemented.Unimplemented
+
+import Standard.Table.Internal.Problem_Builder.Problem_Builder
+import Standard.Table.Internal.Vector_Builder.Vector_Builder
+from Standard.Table import Aggregate_Column, Column, Value_Type
+from Standard.Table.Aggregate_Column.Aggregate_Column import all
+from Standard.Table.Errors import Inexact_Type_Coercion
+from Standard.Table.Internal.Storage import get_storage_for_column
+
+import Standard.Database.Connection.Connection.Connection
+import Standard.Database.DB_Column.DB_Column
+import Standard.Database.DB_Table.DB_Table
+import Standard.Database.Dialect
+import Standard.Database.Internal.Base_Generator
+import Standard.Database.Internal.Common.Database_Distinct_Helper
+import Standard.Database.Internal.Common.Database_Join_Helper
+import Standard.Database.Internal.Error_Mapper.Error_Mapper
+import Standard.Database.Internal.Internals_Access
+import Standard.Database.Internal.IR.Context.Context
+import Standard.Database.Internal.IR.Context.Context_Extension
+import Standard.Database.Internal.IR.From_Spec.From_Spec
+import Standard.Database.Internal.IR.Internal_Column.Internal_Column
+import Standard.Database.Internal.IR.Nulls_Order.Nulls_Order
+import Standard.Database.Internal.IR.Order_Descriptor.Order_Descriptor
+import Standard.Database.Internal.IR.Query.Query
+import Standard.Database.Internal.IR.SQL_Expression.SQL_Expression
+import Standard.Database.Internal.IR.SQL_Join_Kind.SQL_Join_Kind
+## TODO replace with custom one
+import Standard.Database.Internal.Postgres.Postgres_Error_Mapper.Postgres_Error_Mapper
+import Standard.Database.Internal.Replace_Params.Replace_Params
+import Standard.Database.Internal.SQL_Type_Mapping.SQL_Type_Mapping
+import Standard.Database.Internal.SQL_Type_Reference.SQL_Type_Reference
+import Standard.Database.Internal.Statement_Setter.Statement_Setter
+import Standard.Database.SQL.SQL_Builder
+import Standard.Database.SQL.SQL_Fragment
+import Standard.Database.SQL_Statement.SQL_Statement
+import Standard.Database.SQL_Type.SQL_Type
+from Standard.Database.Errors import SQL_Error, Unsupported_Database_Operation
+from Standard.Database.Internal.IR.Operation_Metadata import Date_Period_Metadata
+from Standard.Database.Internal.Statement_Setter import fill_hole_default
+
+import project.Internal.SQLServer_Type_Mapping.SQLServer_Type_Mapping
+
+polyglot java import java.sql.Types as Java_Types
+polyglot java import org.enso.database.JDBCUtils
+
+## PRIVATE
+   The dialect of SQL Server databases.
+sqlserver : SQLSever_Dialect
+sqlserver =
+    SQLSever_Dialect.Value make_dialect_operations
+
+## PRIVATE
+   The dialect of SQL Server databases.
+type SQLSever_Dialect
+    ## PRIVATE
+       The dialect of SQL Server databases.
+    Value dialect_operations
+
+    ## PRIVATE
+       Name of the dialect.
+    name : Text
+    name self = sqlserver_dialect_name
+
+    ## PRIVATE
+    to_text : Text
+    to_text self = "SQLServer_Dialect"
```

Comment:
I'm wondering if it would make sense to call this `Transact_SQL_Dialect`? Or is there a subtle difference between T-SQL and the specifics of SQL Server?

Reply:
Not that I expect we will do it anytime soon but Sybase has a different Transact SQL dialect. I think SQLServer_Dialect is fine but maybe should be SQL_Server.